pkg/segment/reader/segread/segstatsreader.go

```diff
@@ -143,37 +143,52 @@ func readSingleSst(fdata []byte, qid uint64) (*structs.SegStats, error) {
 
 	idx += hllSize
 
+	if version == utils.VERSION_SEGSTATS_BUF_V4[0] {
+		err := readCurrentSST(&sst, fdata, idx)
+		if err != nil {
+			return nil, fmt.Errorf("readSingleSst: error reading current sst, err: %v", err)
+		}
+		return &sst, nil
+	}
+
```

Comment:
nit: add a comment, versions prior to V4 only had numeric stats.
so that we know the code below this if block is for older versions

Reply:
addressed in the latest commit.